packages/valory/skills/decision_maker_abci/behaviours/sampling.py

```diff
@@ -67,7 +72,14 @@ def processable_bet(self, bet: Bet) -> bool:
 
         # if we should not rebet, we have all the information we need
         if not self.should_rebet:
-            return within_ranges
+            # the `has_liquidity_changed` check is dangerous; this can result in a bet never being processed
+            # e.g.:
+            #     1. a market is selected
+            #     2. the mech is uncertain
+            #     3. a bet is not placed
+            #     4. the market's liquidity never changes
+            #     5. the market is never selected again, and therefore a bet is never placed on it
+            return within_ranges and self.has_liquidity_changed(bet)
```

Comment:
has_liquidity_changed could return None here if for some reason the bet_id is not found in the dictionary, should you not check it somewhere?

Reply:
`has_liquidity_changed` can only return a `bool`.